### PR TITLE
Basic markdown renderer

### PR DESCRIPTION
**Pull request by mrdrogdrog**
This PR adds MarkdownIt as dependency and implements a basic editor<=>renderer interaction.

### CODE REVIEW

**Comment by mrdrogdrog** [comment]
This PR is already big enough. We should merge it and then create new feature branches for the missing plugins.

File: src/components/editor/markdown-preview/replace-components/video-util.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { DomElement } from 'domhandler'
+
+export const testSingleVideoParagraph = (node: DomElement, tagName: string): (string | undefined) => {
```
Comment (DerMolly) [inline src/components/editor/markdown-preview/replace-components/video-util.ts:3]: we should rename this probably

File: src/components/editor/markdown-preview/replace-components/vimeo/vimeo-frame.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const videoId = testSingleVideoParagraph(node, 'vimeo')
+  if (videoId) {
+    const count = (counterMap.get(videoId) || 0) + 1
+    counterMap.set(videoId, count)
+    return <VimeoFrame key={`vimeo_${videoId}_${count}`} id={videoId}/>
+  }
+}
+
+interface VimeoApiResponse {
+  // eslint-disable-next-line camelcase
```
Comment (DerMolly) [inline src/components/editor/markdown-preview/replace-components/vimeo/vimeo-frame.tsx:16]: should we add an comment to explain this?

**Comment by DerMolly** [comment]
This PR needs to change `CHANGELOG.md`

File: src/components/editor/markdown-preview/replace-components/video-util.ts (hunk 1 above)
Comment (mrdrogdrog) [reply src/components/editor/markdown-preview/replace-components/video-util.ts]: renamed

File: src/components/editor/markdown-preview/replace-components/vimeo/vimeo-frame.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply src/components/editor/markdown-preview/replace-components/vimeo/vimeo-frame.tsx:17]: comment added

**Pull request merged**